sdk/keyvault/azure_security_keyvault_secrets/README.md

```diff
@@ -1,24 +1,499 @@
+`delete_secret` starts a long-running operation to delete a secret previously stored in the Azure Key Vault. You can retrieve the secret immediately without waiting for the operation to complete. When [soft-delete] is not enabled for the Azure Key Vault, this operation permanently deletes the secret.
+
+```rust
+use azure_core::Response;
+use azure_identity::DefaultAzureCredential;
+use azure_security_keyvault_secrets::{models::DeletedSecretBundle, SecretClient};
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let credential = DefaultAzureCredential::new()?;
+    let client = SecretClient::new(
+        "https://your-key-vault-name.vault.azure.net/",
+        credential,
+        None,
+    )?;
+
+    // Delete a secret using the secret client.
+    let response: Response<DeletedSecretBundle> = client
+        .delete_secret(
+            "secret_name".to_string(),
+            None,
+        ).await?;
+    println!("Response Code: {:?}", response.status());
+
+    Ok(())
+}
+```
+
+### Delete and purge a secret
+
+You will need to wait for the long-running operation to complete before trying to purge or recover the secret. You can do this by calling `update_status` in a loop as shown below:
+
+```rust
+use azure_core::Response;
+use azure_identity::DefaultAzureCredential;
+use azure_security_keyvault_secrets::{models::DeletedSecretBundle, SecretClient};
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let credential = DefaultAzureCredential::new()?;
+    let client = SecretClient::new(
+        "https://your-key-vault-name.vault.azure.net/",
+        credential,
+        None,
+    )?;
+
+    // Delete a secret using the secret client.
+    let response: Response<DeletedSecretBundle> = client
+        .delete_secret("secret_name".to_string(), None)
+        .await?;
+    println!("Delete Response Code: {:?}", response.status());
+
+    // Purge deleted secret using the secret client.
+    let response: Response<()> = client
+        .purge_deleted_secret("secret_name".to_string(), None)
+        .await?;
+    println!("Purge Response Code: {:?}", response.status());
+
+    Ok(())
+}
+
+```
+
+### List secrets
+
+This example lists all the secrets in the specified Azure Key Vault. The value is not returned when listing all secrets. You will need to call `SecretClient::get_secret` to retrieve the value.
+
+```rust
+use azure_core::Pager;
+use azure_identity::DefaultAzureCredential;
+use azure_security_keyvault_secrets::{models::SecretListResult, SecretClient};
+
+#[tokio::main]
+async fn main() -> Result<(), Box<dyn std::error::Error>> {
+    let credential = DefaultAzureCredential::new()?;
+    let client = SecretClient::new(
+        "https://your-key-vault-name.vault.azure.net/",
+        credential,
+        None,
+    )?;
+
+    // List secrets using the secret client.
+    let response: Pager<SecretListResult>= client.get_secrets(None)?;
+    println!("{:?}", response);
```

Comment:
What is the proper way to work with the Pager struct? I saw it has a stream property but it's private. And no other methods were showing through IntelliSense on the struct.

Reply:
Resolved with @heaths's example from the tests